syntax = "proto3";
package datafilter;




import "resource_manage_sdk/model/resource_manage/filter_strategy.proto";
import "resource_manage_sdk/model/resource_manage/filter_strategy_instance.proto";
import "google/protobuf/struct.proto";

/*
ListStrategy请求
*/
message ListStrategyRequest {
    /*
    query
    */
    google.protobuf.Struct query = 1 ;
    /*
    sort
    */
    google.protobuf.Struct sort = 2 ;
}

/*
ListStrategy返回
*/
message ListStrategyResponse {
message List {
    /*
    策略详情
    */
    resource_manage.FilterStrategy strategy = 1 ;
    /*
    策略最后执行详情
    */
    resource_manage.FilterStrategyInstance instance = 2 ;
}
        
    
    /*
    总数
    */
    int32 total = 1 ;
    /*
    数据列表
    */
    repeated List list = 2 ;
}

/*
ListStrategyApi返回
*/
message ListStrategyResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    datafilter.ListStrategyResponse data = 4 ;
}

